### DISCUSSION TITLE
How to Change Style Dynamically

### DISCUSSION THREAD

**Discussion by cxs-mahendra** [primefaces/community#1840] [PrimeReact]
Hello Folks,

I need to change the style property dynamically.
Let's say my primary-color of a button is Red.
Now I need to change its color to say Green on a click of a button.

My finding
**********
I can see that the base theme style is using $variables like below
.p-button {
    color: $buttonTextColor;
    background: $buttonBg;
    border: $buttonBorder;
    ....
    ....

Had this been written as below, I could have changed that dynamically using Java script.
.p-button {
    color: var(--buttonTextColor);
    background: var(--buttonBg);
    border: var(--buttonBorder);
    ....
    ....

Java script code to change color
*********************************
var r = document.querySelector(':root');
var rs = getComputedStyle(r);
r.style.setProperty('--buttonTextColor', 'green');

Please help..
Regards,
Mahendra

**Comment by JL45312** [comment]
could try
const [cn, setCn] = useState<>("p-button-red")
 <Button className={cn} onClick=(() => setCn("p-button-green")) />

or

let cn  = "p-button-red"
 <Button className={cn} onClick=(() => cn = "p-button-green") />
